packages/parser/src/dast-normalize/normalize-dast.ts

```diff
@@ -160,6 +161,120 @@ const pluginAddCompatibilityNames: Plugin<[], DastRoot, DastRoot> = () => {
     };
 };
 
+/**
+ * Perform substitutions for syntactic sugar based on the component type
+ */
+const pluginComponentSugar: Plugin<[], DastRoot, DastRoot> = () => {
+    return (tree) => {
+        visit(tree, (node) => {
+            if (!isDastElement(node)) {
+                return;
+            }
+            if (node.name === "repeat") {
+                // If a `<repeat>` element has a `"valueName"` and/or `"indexName" attribute with a single text child,
+                // then add a `<_repeatSetup>` child to the `<repeat>` that contains children
+                // named by the values of those attributes.
+                // These children will not be rendered, but they create targets for references to `valueName` and `indexName`.
+                // Mapping those references to the correct target will be addressed when the `<repeat>` is expanded.
+                let setupChildren: DastElementContent[] = [];
+                if (node.attributes.valueName) {
+                    const attrChildren = node.attributes.valueName.children;
+                    if (
+                        attrChildren.length === 1 &&
+                        attrChildren[0].type === "text"
+                    ) {
+                        const valueName = attrChildren[0].value;
+
+                        setupChildren.push({
+                            type: "element",
+                            name: "_placeholder",
```

Comment:
If we're doing stuff like this a lot, it would make sense to have constructor functions, but for now I think its fine to specify the object explicitly.